app/Http/Requests/Company/UpdateCompanyRequest.php

```diff
@@ -67,6 +68,8 @@ public function rules()
             $rules['subdomain'] = ['nullable', 'regex:/^[a-zA-Z0-9.-]+[a-zA-Z0-9]$/', new ValidSubdomain()];
         }
 
+        $rules['expense_mailbox'] = new ValidExpenseMailbox($this->company->key, $this->company->account->isPaid() && $this->company->account->plan == 'enterprise'); // @turbo124 check if this is right
```

Comment:
The best way to handle this would be to place the authorization check in the prepareForValidation. 

ie

```php
public function prepareForValidation() {

if(isset($input['expense_mailbox']) && Ninja::isHosted() && !( $user_account->isPaid() && $user_account->plan == 'enterprise'){
unset($input['expense_mailbox']);
}


}
```

Reply:
implemented